service/history/timerQueueStandbyProcessor.go

```diff
@@ -97,7 +98,9 @@ func newTimerQueueStandbyProcessor(
 		config,
 	)
 
-	scheduler := newTimerTaskScheduler(shard, logger)
+	if scheduler == nil {
+		scheduler = newTimerTaskScheduler(shard, logger)
+	}
```

Comment:
this is only to make unit test happy?

Reply:
It's for falling back to using shard level scheduler if host level scheduler is disabled (the default).